afs/media/js/views/components/workflows/create-project-workflow/add-things-step.js

```diff
@@ -292,11 +164,54 @@ define([
                 data: {
                     'nodeid': collectionNameNodeId,
                     'data':  ("Collection for " + researchActivityName),
-                    'tileid': null,
-                    'resourceinstanceid': ko.unwrap(self.collectionResourceId)
-                }
+                    'tileid': ko.unwrap(self.collectionTileId),
+                    'resourceinstanceid': ko.unwrap(self.collectionResourceId),
+                    'transaction_id': params.form.workflowId
+                },
             }).done(function(data) {
                 self.collectionResourceId(data.resourceinstance_id);
+                self.collectionTileId(data.tileid);
+
+                self.removedValues(self.startValue().filter(val => !self.value().includes(val)));
+                self.addedValues(self.value().filter(val => !self.startValue().includes(val)));
+
+                createActivityUsedSet().then(
```

Comment:
async/await could be used to reduce nesting (since Promise.all returns another promise, as does createActivityUsedSet)

Reply:
I think this would be a nice refactor and definitely worth noting, but we can probably go ahead and merge and return to this later.